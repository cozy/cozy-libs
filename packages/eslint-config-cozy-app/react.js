'use strict'

const basics = require('./basics')

module.exports = {
  plugins: basics.plugins.concat(['react-hooks']),
  extends: basics.extends.concat(['plugin:react/recommended']),
  parser: basics.parser,
  parserOptions: { ecmaFeatures: { jsx: true } },
  env: basics.env,
  settings: { react: { version: 'latest' } },
  rules: Object.assign({}, basics.rules, {
    'react/prop-types': 'off',
    'react-hooks/rules-of-hooks': 'error',
    'react-hooks/exhaustive-deps': 'warn'
  })
}
