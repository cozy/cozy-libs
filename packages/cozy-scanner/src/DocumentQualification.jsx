import React, { Component } from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

import Icon from 'cozy-ui/transpiled/react/Icon'
import { Title } from 'cozy-ui/transpiled/react/Text'
import Label from 'cozy-ui/transpiled/react/Label'
import Input from 'cozy-ui/transpiled/react/Input'
import InputGroup from 'cozy-ui/transpiled/react/InputGroup'
import { translate } from 'cozy-ui/transpiled/react/I18n'
import MuiCozyTheme from 'cozy-ui/transpiled/react/MuiCozyTheme'
import Grid from 'cozy-ui/transpiled/react/MuiCozyTheme/Grid'

import CategoryGridItem from './CategoryGridItem'
import DocumentCategory from './DocumentCategory'

import { themes } from './DocumentTypeData'
import { getItemById, getItemsByCategory } from './DocumentTypeDataHelpers'
import GridItem from 'cozy-ui/transpiled/react/Labs/GridItem'
import styles from './stylesheet.css'

const fileExtension = '.jpg'
const idFileInput = 'filename_input'

/**
 *
 * Used to describe a document ie:
 *  - Selecting a metadata category etc
 *  - Renaming the file
 *
 * When a selection is done,
 * `props.onDescribed` is called with the selected item
 *
 */
export class DocumentQualification extends Component {
  constructor(props) {
    super(props)
    const { categoryLabel = null, itemId = null } = props.initialSelected || {}
    this.defaultFilename = `Scan_${new Date().toISOString().replace(/:/g, '-')}`
    this.state = {
      selected: { categoryLabel, itemId },
      filename: this.defaultFilename,
      hasUserWrittenFileName: false
    }
    this.textInput = React.createRef()
  }

  getFilenameFromCategory = (item, t) => {
    const realItem = getItemById(item.itemId)
    if (realItem) {
      const name = t(`Scan.items.${realItem.label}`)
      return `${name.replace(/ /g, '-')}_${new Date()
        .toISOString()
        .substr(0, 10)}`
    } else {
      return this.defaultFilename
    }
  }

  onSelect = item => {
    const { t, allowEditFileName } = this.props
    const { hasUserWrittenFileName } = this.state
    let filename = null
    if (!hasUserWrittenFileName) {
      filename = this.getFilenameFromCategory(item, t)
    } else {
      filename = this.state.filename
    }
    this.setState({ selected: item })
    this.handleFileNameChange(filename)

    const { onDescribed } = this.props
    if (onDescribed) {
      const realItem = getItemById(item.itemId)
      /* ATM, we only accept JPG extension from the scanner. So
      we hardcode the filename extension here.

      If we can't edit file name, then we don't need to use the
      generated filename
      */
      if (allowEditFileName) {
        onDescribed(realItem ? realItem : undefined, filename + fileExtension)
      } else {
        onDescribed(realItem ? realItem : undefined)
      }
    }
  }
  /**
   * Method used to synchronize our internal state and
   * our parent state if needed
   */
  handleFileNameChange = filename => {
    const { onFileNameChanged } = this.props
    this.setState({ filename })
    onFileNameChanged && onFileNameChanged(filename + fileExtension)
  }

  render() {
    const { t, title, allowEditFileName } = this.props
    const { selected, filename, hasUserWrittenFileName } = this.state
    return (
      <MuiCozyTheme>
        {allowEditFileName && (
          <>
            <Label htmlFor={idFileInput}>{t('Scan.filename')}</Label>
            <InputGroup
              fullwidth
              append={<span className="u-pr-1">{fileExtension}</span>}
              className="u-bdrs-3"
            >
              <Input
                placeholder={t('Scan.filename')}
                value={filename}
                onChange={event => {
                  // console.log('event', event)
                  // If the user write something once, we don't want to rename the file automatically anymore
                  if (!hasUserWrittenFileName) {
                    this.setState({ hasUserWrittenFileName: true })
                  }
                  // If we left an empty value, then we reset the behavior
                  if (event.target.value === '') {
                    this.setState({ hasUserWrittenFileName: false })
                  }

                  this.handleFileNameChange(event.target.value)
                }}
                onFocus={() => {
                  if (!hasUserWrittenFileName)
                    this.textInput.current.setSelectionRange(0, filename.length)
                }}
                onBlur={() => {
                  if (filename === '') {
                    this.handleFileNameChange(
                      this.getFilenameFromCategory(selected, t)
                    )
                  }
                }}
                inputRef={this.textInput}
                id={idFileInput}
              />
            </InputGroup>
          </>
        )}

        {title && (
          <div className="u-flex u-flex-items-center u-mt-1-half">
            <Icon icon="qualify" />
            <Title
              className={classNames(
                styles['grid-item-title'],
                'u-pl-half u-mb-1'
              )}
            >
              {title}
            </Title>
          </div>
        )}
        <Grid container spacing={8}>
          <GridItem
            onClick={() => this.onSelect({ categoryLabel: null, itemId: null })}
          >
            <CategoryGridItem
              isSelected={selected.categoryLabel === null}
              theme={t(`Scan.themes.undefined`)}
            />
          </GridItem>

          {themes.map((category, i) => {
            return (
              <DocumentCategory
                onSelect={item => this.onSelect(item)}
                category={category}
                items={getItemsByCategory(category)}
                key={i}
                isSelected={selected.categoryLabel === category.label}
                selectedItem={
                  selected.itemId ? getItemById(selected.itemId) : {}
                }
                t={t}
              />
            )
          })}
        </Grid>
      </MuiCozyTheme>
    )
  }
}

DocumentQualification.defaultProps = {
  allowEditFileName: false
}
DocumentQualification.propTypes = {
  /** This callback is called after a qualification with
   * the selected item
   */
  onDescribed: PropTypes.func,
  onFileNameChanged: PropTypes.func,
  title: PropTypes.string,
  initialSelected: PropTypes.shape({
    itemId: PropTypes.string,
    categoryLabel: PropTypes.string
  }),
  /** define is the user can edit the filename */
  allowEditFileName: PropTypes.bool
}
export default translate()(DocumentQualification)
