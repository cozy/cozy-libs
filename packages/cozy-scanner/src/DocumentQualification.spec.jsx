import React from 'react'
const MockDate = require('mockdate')
import { render, fireEvent } from '@testing-library/react'
jest.mock('cozy-ui/transpiled/react/utils/color', () => ({
  getCssVariableValue: () => '#fff'
}))
// Popper does not work well inside of jest as it heavily relies on DOM APIs (see https://github.com/popperjs/popper-core/issues/478).
jest.mock('@material-ui/core/Popper', () => {
  return ({ children }) => children
})

import { DocumentQualification } from './DocumentQualification'
import MuiCozyTheme from 'cozy-ui/transpiled/react/MuiCozyTheme'

const MOCKED_DATE = '2019-01-01'

describe('DocumentQualification', () => {
  beforeAll(() => {
    MockDate.set(MOCKED_DATE)
  })

  afterAll(() => {
    MockDate.reset()
  })
  it('Initial render + selection + filename', () => {
    const onDescribed = jest.fn()
    const onFileNameChanged = jest.fn()
    const { queryByText, getByText, asFragment, getByLabelText } = render(
      <MuiCozyTheme>
        <DocumentQualification
          allowEditFileName={true}
          onDescribed={onDescribed}
          onFileNameChanged={onFileNameChanged}
          title={'Edit'}
          // initialSelected={{ itemId: 1, categoryLabel: 'Label1' }}
          t={text => text}
        />
      </MuiCozyTheme>
    )
    expect(asFragment()).toMatchSnapshot()
    const undefinedItemNode = queryByText(/Scan.themes.undefined/i)
    const isSelected = undefinedItemNode.closest(
      '.grid-item__selected__without_label'
    )
    expect(Boolean(isSelected)).toBe(true)

    fireEvent.click(getByText('Scan.themes.family'))
    fireEvent.click(getByText('Scan.items.family_record_book'))
    expect(onDescribed).toBeCalledWith(
      {
        classification: 'identity_document',
        defaultTheme: 'theme2',
        id: '4',
        label: 'family_record_book',
        subClassification: 'family_record_book'
      },
      'Scan.items.family_record_book_2019-01-01.jpg'
    )
    const inputFileName = getByLabelText('Scan.filename')

    const familyItemNode = queryByText('Scan.themes.family')
    const familyItemNodeSelected = familyItemNode.closest(
      '.grid-item__selected'
    )
    expect(Boolean(familyItemNodeSelected)).toBe(true)
    expect(inputFileName.value).toBe('Scan.items.family_record_book_2019-01-01')
    expect(asFragment()).toMatchSnapshot()

    fireEvent.change(inputFileName, { target: { value: '' } })
    fireEvent.blur(inputFileName)

    expect(inputFileName.value).toBe('Scan.items.family_record_book_2019-01-01')

    fireEvent.change(inputFileName, { target: { value: 'Manual name' } })
    fireEvent.blur(inputFileName)
    expect(onFileNameChanged).toBeCalledWith('Manual name.jpg')

    fireEvent.click(familyItemNode)
    fireEvent.click(queryByText('Scan.items.birth_certificate'))
    expect(inputFileName.value).toBe('Manual name')
  })
})
