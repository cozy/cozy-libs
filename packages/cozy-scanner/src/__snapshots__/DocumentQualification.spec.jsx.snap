// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentQualification Inital render + selection + filename 1`] = `
<DocumentFragment>
  <label
    class="styles__c-label___o4ozG styles__c-label--block___2ZV_7"
    for="filename_input"
  >
    Scan.filename
  </label>
  <div
    class="styles__c-inputgroup___12OVJ styles__c-inputgroup--fullwidth___3nuay u-bdrs-3"
  >
    <div
      class="styles__c-inputgroup-main___1LP4B"
    >
      <input
        class="styles__c-input-text___3TAv1 styles__c-input-text--large___28EaR"
        id="filename_input"
        placeholder="Scan.filename"
        type="text"
        value="Scan_2019-01-01T00-00-00.000Z"
      />
    </div>
    <div
      class="styles__c-inputgroup-side___60v0v"
    >
      <span
        class="u-pr-1"
      >
        .jpg
      </span>
    </div>
  </div>
  <div
    class="u-flex u-flex-items-center u-mt-1-half"
  >
    <svg
      class="styles__icon___23x3R"
      height="16"
      width="16"
    >
      <use
        xlink:href="#qualify"
      />
    </svg>
    <div
      class="u-title-h2 grid-item-title u-pl-half u-mb-1"
    >
      Edit
    </div>
  </div>
  <div
    class="MuiGrid-container-1"
  >
    <div
      class="MuiGrid-item-2 MuiGrid-grid-xs-3-32 MuiGrid-grid-sm-2-45 styles__gridItem-container___387sO"
    >
      <div
        class="grid-item u-pb-half u-ph-half u-bxz  u-ellipsis u-c-pointer border-selected grid-item__selected__without_label"
      >
        <div
          class="styles__IconStack-wrapper___10dhG"
        >
          <svg
            class="styles__icon___23x3R"
            height="32"
            style="fill: var(--dodgerBlue);"
            width="32"
          >
            <use
              xlink:href="#file-duotone"
            />
          </svg>
          <div
            class="styles__IconStack-foregroundIcon___ZvY-t"
          />
        </div>
        <div
          class="u-flex u-flex-column u-ellipsis u-bxz"
        >
          <span
            class="grid-item-theme u-ellipsis"
          >
            Scan.themes.undefined
          </span>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-item-2 MuiGrid-grid-xs-3-32 MuiGrid-grid-sm-2-45 styles__gridItem-container___387sO"
    >
      <div
        class="grid-item u-pb-half u-ph-half u-bxz  u-ellipsis u-c-pointer border-not-selected"
      >
        <div
          class="styles__IconStack-wrapper___10dhG"
        >
          <svg
            class="styles__icon___23x3R"
            height="32"
            style="fill: var(--coolGrey);"
            width="32"
          >
            <use
              xlink:href="#file-duotone"
            />
          </svg>
          <div
            class="styles__IconStack-foregroundIcon___ZvY-t"
          >
            <svg
              class="styles__icon___23x3R"
              height="16"
              style="fill: var(--coolGrey);"
              width="16"
            >
              <use
                xlink:href="#people"
              />
            </svg>
          </div>
        </div>
        <div
          class="u-flex u-flex-column u-ellipsis u-bxz"
        >
          <span
            class="grid-item-theme u-ellipsis"
          >
            Scan.themes.identity
          </span>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-item-2 MuiGrid-grid-xs-3-32 MuiGrid-grid-sm-2-45 styles__gridItem-container___387sO"
    >
      <div
        class="grid-item u-pb-half u-ph-half u-bxz  u-ellipsis u-c-pointer border-not-selected"
      >
        <div
          class="styles__IconStack-wrapper___10dhG"
        >
          <svg
            class="styles__icon___23x3R"
            height="32"
            style="fill: var(--coolGrey);"
            width="32"
          >
            <use
              xlink:href="#file-duotone"
            />
          </svg>
          <div
            class="styles__IconStack-foregroundIcon___ZvY-t"
          >
            <svg
              class="styles__icon___23x3R"
              height="16"
              style="fill: var(--coolGrey);"
              width="16"
            >
              <use
                xlink:href="#team"
              />
            </svg>
          </div>
        </div>
        <div
          class="u-flex u-flex-column u-ellipsis u-bxz"
        >
          <span
            class="grid-item-theme u-ellipsis"
          >
            Scan.themes.family
          </span>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-item-2 MuiGrid-grid-xs-3-32 MuiGrid-grid-sm-2-45 styles__gridItem-container___387sO"
    >
      <div
        class="grid-item u-pb-half u-ph-half u-bxz  u-ellipsis u-c-pointer border-not-selected"
      >
        <div
          class="styles__IconStack-wrapper___10dhG"
        >
          <svg
            class="styles__icon___23x3R"
            height="32"
            style="fill: var(--coolGrey);"
            width="32"
          >
            <use
              xlink:href="#file-duotone"
            />
          </svg>
          <div
            class="styles__IconStack-foregroundIcon___ZvY-t"
          >
            <svg
              class="styles__icon___23x3R"
              height="16"
              style="fill: var(--coolGrey);"
              width="16"
            >
              <use
                xlink:href="#company"
              />
            </svg>
          </div>
        </div>
        <div
          class="u-flex u-flex-column u-ellipsis u-bxz"
        >
          <span
            class="grid-item-theme u-ellipsis"
          >
            Scan.themes.work_study
          </span>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-item-2 MuiGrid-grid-xs-3-32 MuiGrid-grid-sm-2-45 styles__gridItem-container___387sO"
    >
      <div
        class="grid-item u-pb-half u-ph-half u-bxz  u-ellipsis u-c-pointer border-not-selected"
      >
        <div
          class="styles__IconStack-wrapper___10dhG"
        >
          <svg
            class="styles__icon___23x3R"
            height="32"
            style="fill: var(--coolGrey);"
            width="32"
          >
            <use
              xlink:href="#file-duotone"
            />
          </svg>
          <div
            class="styles__IconStack-foregroundIcon___ZvY-t"
          >
            <svg
              class="styles__icon___23x3R"
              height="16"
              style="fill: var(--coolGrey);"
              width="16"
            >
              <use
                xlink:href="#heart"
              />
            </svg>
          </div>
        </div>
        <div
          class="u-flex u-flex-column u-ellipsis u-bxz"
        >
          <span
            class="grid-item-theme u-ellipsis"
          >
            Scan.themes.health
          </span>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-item-2 MuiGrid-grid-xs-3-32 MuiGrid-grid-sm-2-45 styles__gridItem-container___387sO"
    >
      <div
        class="grid-item u-pb-half u-ph-half u-bxz  u-ellipsis u-c-pointer border-not-selected"
      >
        <div
          class="styles__IconStack-wrapper___10dhG"
        >
          <svg
            class="styles__icon___23x3R"
            height="32"
            style="fill: var(--coolGrey);"
            width="32"
          >
            <use
              xlink:href="#file-duotone"
            />
          </svg>
          <div
            class="styles__IconStack-foregroundIcon___ZvY-t"
          >
            <svg
              class="styles__icon___23x3R"
              height="16"
              style="fill: var(--coolGrey);"
              width="16"
            >
              <use
                xlink:href="#home"
              />
            </svg>
          </div>
        </div>
        <div
          class="u-flex u-flex-column u-ellipsis u-bxz"
        >
          <span
            class="grid-item-theme u-ellipsis"
          >
            Scan.themes.home
          </span>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-item-2 MuiGrid-grid-xs-3-32 MuiGrid-grid-sm-2-45 styles__gridItem-container___387sO"
    >
      <div
        class="grid-item u-pb-half u-ph-half u-bxz  u-ellipsis u-c-pointer border-not-selected"
      >
        <div
          class="styles__IconStack-wrapper___10dhG"
        >
          <svg
            class="styles__icon___23x3R"
            height="32"
            style="fill: var(--coolGrey);"
            width="32"
          >
            <use
              xlink:href="#file-duotone"
            />
          </svg>
          <div
            class="styles__IconStack-foregroundIcon___ZvY-t"
          >
            <svg
              class="styles__icon___23x3R"
              height="16"
              style="fill: var(--coolGrey);"
              width="16"
            >
              <use
                xlink:href="#car"
              />
            </svg>
          </div>
        </div>
        <div
          class="u-flex u-flex-column u-ellipsis u-bxz"
        >
          <span
            class="grid-item-theme u-ellipsis"
          >
            Scan.themes.transport
          </span>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-item-2 MuiGrid-grid-xs-3-32 MuiGrid-grid-sm-2-45 styles__gridItem-container___387sO"
    >
      <div
        class="grid-item u-pb-half u-ph-half u-bxz  u-ellipsis u-c-pointer border-not-selected"
      >
        <div
          class="styles__IconStack-wrapper___10dhG"
        >
          <svg
            class="styles__icon___23x3R"
            height="32"
            style="fill: var(--coolGrey);"
            width="32"
          >
            <use
              xlink:href="#file-duotone"
            />
          </svg>
          <div
            class="styles__IconStack-foregroundIcon___ZvY-t"
          >
            <svg
              class="styles__icon___23x3R"
              height="16"
              style="fill: var(--coolGrey);"
              width="16"
            >
              <use
                xlink:href="#bank"
              />
            </svg>
          </div>
        </div>
        <div
          class="u-flex u-flex-column u-ellipsis u-bxz"
        >
          <span
            class="grid-item-theme u-ellipsis"
          >
            Scan.themes.tax
          </span>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-item-2 MuiGrid-grid-xs-3-32 MuiGrid-grid-sm-2-45 styles__gridItem-container___387sO"
    >
      <div
        class="grid-item u-pb-half u-ph-half u-bxz  u-ellipsis u-c-pointer border-not-selected"
      >
        <div
          class="styles__IconStack-wrapper___10dhG"
        >
          <svg
            class="styles__icon___23x3R"
            height="32"
            style="fill: var(--coolGrey);"
            width="32"
          >
            <use
              xlink:href="#file-duotone"
            />
          </svg>
          <div
            class="styles__IconStack-foregroundIcon___ZvY-t"
          >
            <svg
              class="styles__icon___23x3R"
              height="16"
              style="fill: var(--coolGrey);"
              width="16"
            >
              <use
                xlink:href="#euro"
              />
            </svg>
          </div>
        </div>
        <div
          class="u-flex u-flex-column u-ellipsis u-bxz"
        >
          <span
            class="grid-item-theme u-ellipsis"
          >
            Scan.themes.invoice
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`DocumentQualification Inital render + selection + filename 2`] = `
<DocumentFragment>
  <label
    class="styles__c-label___o4ozG styles__c-label--block___2ZV_7"
    for="filename_input"
  >
    Scan.filename
  </label>
  <div
    class="styles__c-inputgroup___12OVJ styles__c-inputgroup--fullwidth___3nuay u-bdrs-3"
  >
    <div
      class="styles__c-inputgroup-main___1LP4B"
    >
      <input
        class="styles__c-input-text___3TAv1 styles__c-input-text--large___28EaR"
        id="filename_input"
        placeholder="Scan.filename"
        type="text"
        value="Scan.items.family_record_book_2019-01-01"
      />
    </div>
    <div
      class="styles__c-inputgroup-side___60v0v"
    >
      <span
        class="u-pr-1"
      >
        .jpg
      </span>
    </div>
  </div>
  <div
    class="u-flex u-flex-items-center u-mt-1-half"
  >
    <svg
      class="styles__icon___23x3R"
      height="16"
      width="16"
    >
      <use
        xlink:href="#qualify"
      />
    </svg>
    <div
      class="u-title-h2 grid-item-title u-pl-half u-mb-1"
    >
      Edit
    </div>
  </div>
  <div
    class="MuiGrid-container-1"
  >
    <div
      class="MuiGrid-item-2 MuiGrid-grid-xs-3-32 MuiGrid-grid-sm-2-45 styles__gridItem-container___387sO"
    >
      <div
        class="grid-item u-pb-half u-ph-half u-bxz  u-ellipsis u-c-pointer border-not-selected"
      >
        <div
          class="styles__IconStack-wrapper___10dhG"
        >
          <svg
            class="styles__icon___23x3R"
            height="32"
            style="fill: var(--coolGrey);"
            width="32"
          >
            <use
              xlink:href="#file-duotone"
            />
          </svg>
          <div
            class="styles__IconStack-foregroundIcon___ZvY-t"
          />
        </div>
        <div
          class="u-flex u-flex-column u-ellipsis u-bxz"
        >
          <span
            class="grid-item-theme u-ellipsis"
          >
            Scan.themes.undefined
          </span>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-item-2 MuiGrid-grid-xs-3-32 MuiGrid-grid-sm-2-45 styles__gridItem-container___387sO"
    >
      <div
        class="grid-item u-pb-half u-ph-half u-bxz  u-ellipsis u-c-pointer border-not-selected"
      >
        <div
          class="styles__IconStack-wrapper___10dhG"
        >
          <svg
            class="styles__icon___23x3R"
            height="32"
            style="fill: var(--coolGrey);"
            width="32"
          >
            <use
              xlink:href="#file-duotone"
            />
          </svg>
          <div
            class="styles__IconStack-foregroundIcon___ZvY-t"
          >
            <svg
              class="styles__icon___23x3R"
              height="16"
              style="fill: var(--coolGrey);"
              width="16"
            >
              <use
                xlink:href="#people"
              />
            </svg>
          </div>
        </div>
        <div
          class="u-flex u-flex-column u-ellipsis u-bxz"
        >
          <span
            class="grid-item-theme u-ellipsis"
          >
            Scan.themes.identity
          </span>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-item-2 MuiGrid-grid-xs-3-32 MuiGrid-grid-sm-2-45 styles__gridItem-container___387sO"
    >
      <div
        class="grid-item u-pb-half u-ph-half u-bxz  u-ellipsis u-c-pointer border-selected grid-item__selected"
      >
        <div
          class="styles__IconStack-wrapper___10dhG"
        >
          <svg
            class="styles__icon___23x3R"
            height="32"
            style="fill: var(--dodgerBlue);"
            width="32"
          >
            <use
              xlink:href="#file-duotone"
            />
          </svg>
          <div
            class="styles__IconStack-foregroundIcon___ZvY-t"
          >
            <svg
              class="styles__icon___23x3R"
              height="16"
              style="fill: var(--dodgerBlue);"
              width="16"
            >
              <use
                xlink:href="#team"
              />
            </svg>
          </div>
        </div>
        <div
          class="u-flex u-flex-column u-ellipsis u-bxz"
        >
          <span
            class="grid-item-theme u-ellipsis"
          >
            Scan.themes.family
          </span>
          <span
            class="grid-item-label u-ellipsis"
          >
            Scan.items.family_record_book
          </span>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-item-2 MuiGrid-grid-xs-3-32 MuiGrid-grid-sm-2-45 styles__gridItem-container___387sO"
    >
      <div
        class="grid-item u-pb-half u-ph-half u-bxz  u-ellipsis u-c-pointer border-not-selected"
      >
        <div
          class="styles__IconStack-wrapper___10dhG"
        >
          <svg
            class="styles__icon___23x3R"
            height="32"
            style="fill: var(--coolGrey);"
            width="32"
          >
            <use
              xlink:href="#file-duotone"
            />
          </svg>
          <div
            class="styles__IconStack-foregroundIcon___ZvY-t"
          >
            <svg
              class="styles__icon___23x3R"
              height="16"
              style="fill: var(--coolGrey);"
              width="16"
            >
              <use
                xlink:href="#company"
              />
            </svg>
          </div>
        </div>
        <div
          class="u-flex u-flex-column u-ellipsis u-bxz"
        >
          <span
            class="grid-item-theme u-ellipsis"
          >
            Scan.themes.work_study
          </span>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-item-2 MuiGrid-grid-xs-3-32 MuiGrid-grid-sm-2-45 styles__gridItem-container___387sO"
    >
      <div
        class="grid-item u-pb-half u-ph-half u-bxz  u-ellipsis u-c-pointer border-not-selected"
      >
        <div
          class="styles__IconStack-wrapper___10dhG"
        >
          <svg
            class="styles__icon___23x3R"
            height="32"
            style="fill: var(--coolGrey);"
            width="32"
          >
            <use
              xlink:href="#file-duotone"
            />
          </svg>
          <div
            class="styles__IconStack-foregroundIcon___ZvY-t"
          >
            <svg
              class="styles__icon___23x3R"
              height="16"
              style="fill: var(--coolGrey);"
              width="16"
            >
              <use
                xlink:href="#heart"
              />
            </svg>
          </div>
        </div>
        <div
          class="u-flex u-flex-column u-ellipsis u-bxz"
        >
          <span
            class="grid-item-theme u-ellipsis"
          >
            Scan.themes.health
          </span>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-item-2 MuiGrid-grid-xs-3-32 MuiGrid-grid-sm-2-45 styles__gridItem-container___387sO"
    >
      <div
        class="grid-item u-pb-half u-ph-half u-bxz  u-ellipsis u-c-pointer border-not-selected"
      >
        <div
          class="styles__IconStack-wrapper___10dhG"
        >
          <svg
            class="styles__icon___23x3R"
            height="32"
            style="fill: var(--coolGrey);"
            width="32"
          >
            <use
              xlink:href="#file-duotone"
            />
          </svg>
          <div
            class="styles__IconStack-foregroundIcon___ZvY-t"
          >
            <svg
              class="styles__icon___23x3R"
              height="16"
              style="fill: var(--coolGrey);"
              width="16"
            >
              <use
                xlink:href="#home"
              />
            </svg>
          </div>
        </div>
        <div
          class="u-flex u-flex-column u-ellipsis u-bxz"
        >
          <span
            class="grid-item-theme u-ellipsis"
          >
            Scan.themes.home
          </span>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-item-2 MuiGrid-grid-xs-3-32 MuiGrid-grid-sm-2-45 styles__gridItem-container___387sO"
    >
      <div
        class="grid-item u-pb-half u-ph-half u-bxz  u-ellipsis u-c-pointer border-not-selected"
      >
        <div
          class="styles__IconStack-wrapper___10dhG"
        >
          <svg
            class="styles__icon___23x3R"
            height="32"
            style="fill: var(--coolGrey);"
            width="32"
          >
            <use
              xlink:href="#file-duotone"
            />
          </svg>
          <div
            class="styles__IconStack-foregroundIcon___ZvY-t"
          >
            <svg
              class="styles__icon___23x3R"
              height="16"
              style="fill: var(--coolGrey);"
              width="16"
            >
              <use
                xlink:href="#car"
              />
            </svg>
          </div>
        </div>
        <div
          class="u-flex u-flex-column u-ellipsis u-bxz"
        >
          <span
            class="grid-item-theme u-ellipsis"
          >
            Scan.themes.transport
          </span>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-item-2 MuiGrid-grid-xs-3-32 MuiGrid-grid-sm-2-45 styles__gridItem-container___387sO"
    >
      <div
        class="grid-item u-pb-half u-ph-half u-bxz  u-ellipsis u-c-pointer border-not-selected"
      >
        <div
          class="styles__IconStack-wrapper___10dhG"
        >
          <svg
            class="styles__icon___23x3R"
            height="32"
            style="fill: var(--coolGrey);"
            width="32"
          >
            <use
              xlink:href="#file-duotone"
            />
          </svg>
          <div
            class="styles__IconStack-foregroundIcon___ZvY-t"
          >
            <svg
              class="styles__icon___23x3R"
              height="16"
              style="fill: var(--coolGrey);"
              width="16"
            >
              <use
                xlink:href="#bank"
              />
            </svg>
          </div>
        </div>
        <div
          class="u-flex u-flex-column u-ellipsis u-bxz"
        >
          <span
            class="grid-item-theme u-ellipsis"
          >
            Scan.themes.tax
          </span>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-item-2 MuiGrid-grid-xs-3-32 MuiGrid-grid-sm-2-45 styles__gridItem-container___387sO"
    >
      <div
        class="grid-item u-pb-half u-ph-half u-bxz  u-ellipsis u-c-pointer border-not-selected"
      >
        <div
          class="styles__IconStack-wrapper___10dhG"
        >
          <svg
            class="styles__icon___23x3R"
            height="32"
            style="fill: var(--coolGrey);"
            width="32"
          >
            <use
              xlink:href="#file-duotone"
            />
          </svg>
          <div
            class="styles__IconStack-foregroundIcon___ZvY-t"
          >
            <svg
              class="styles__icon___23x3R"
              height="16"
              style="fill: var(--coolGrey);"
              width="16"
            >
              <use
                xlink:href="#euro"
              />
            </svg>
          </div>
        </div>
        <div
          class="u-flex u-flex-column u-ellipsis u-bxz"
        >
          <span
            class="grid-item-theme u-ellipsis"
          >
            Scan.themes.invoice
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
